unity/MagicLeapFaceDetectionPOC/Assets/Scripts/CompanionDevice/CompanionDevice.cs

```diff
@@ -0,0 +1,479 @@
+      Logger.D("XXX", "Registering Device uid: " + ConfiguredDeviceRegisterRequest.user_id);
+      Logger.D("XXX", "Registering Device uuid: " + ConfiguredDeviceRegisterRequest.device_uuid);
+      */
+      
+      DataContractJsonSerializer serializer = new DataContractJsonSerializer(typeof(DeviceRegisterRequest));
+      Logger.D(TAG, "created serializer.");
+      MemoryStream ms = new MemoryStream();
+      Logger.D(TAG, "created memstream.");
+      serializer.WriteObject(ms, ConfiguredDeviceRegisterRequest);
+      Logger.D(TAG, "wrote object.");
+      string jsonStr = Util.StreamToString(ms);
+
+      string uri = "http://" + host + ":" + port + registerDeviceAPI;
+      Logger.D(TAG, "To this host: " + uri);
+      Stream responseStream = await PostRequest(uri, jsonStr).ConfigureAwait(false);
+
+      if (responseStream == null || !responseStream.CanRead)
+      {
+        throw new EmptyStreamException("DeviceRegisterReply stream is empty!");
+      }
+
+      DataContractJsonSerializer deserializer = new DataContractJsonSerializer(typeof(DeviceRegisterReply));
+      string responseStr = Util.StreamToString(responseStream);
+      byte[] byteArray = Encoding.ASCII.GetBytes(responseStr);
+      ms = new MemoryStream(byteArray);
+      var reply = (DeviceRegisterReply)deserializer.ReadObject(ms);
+
+      if (reply != null)
+      {
+        LastDeviceRegisterReply = reply;
+      }
+
+      return reply;
+    }
+
+    /**
+     * Returns the GPS location from the paired CompanionApp phone/device.
+     */
+    public async Task<LocationReply> GetLocation()
+    {
+      Logger.D(TAG, "GetLocation()");
+
+      if (!MLLocation.IsStarted)
+      {
+        Logger.D(TAG, "Please start MagicLeap's MLLocation object from a GameObject/MonoBehavior before using this call.");
+        // Hm, Debug: return something.
+        Logger.D(TAG, "FIXME: Returning a hard coded value...");
+        return new LocationReply
+        {
+          longitude = -121.955238,
+          latitude = 37.354107
+        };
+      }
+
+      MLLocation.Location locData = new MLLocation.Location();
+      MLResult result = MLLocation.GetLastFineLocation(out locData);
+      Logger.D(TAG, "MLLocation result: " + result); // Ensure location is allowed.
+      if (!result.Equals(MLResult.Create(MLResult.Code.Ok)))
+      {
+        // Try coarse instead:
+        result = MLLocation.GetLastCoarseLocation(out locData);
+        if (!result.Equals(MLResult.Create(MLResult.Code.Ok)))
+        {
+          Logger.D(TAG, "Empty location returning");
+          return new LocationReply { longitude = 0f, latitude = 0f };
+        }
+      }
+
+      Logger.D(TAG, "New location returning: {" + locData.Longitude + ", " + locData.Latitude + "}");
+      return new LocationReply
+      {
+        longitude = locData.Longitude,
+        latitude = locData.Latitude
+      };
+    }
+
+    // From CompanionApp.
+#if false
```

Comment:
Is there a reason for not using (if())?

Reply:
I guess this is for [ConditionalAttribute("DEBUG")] , correct ?

Reply:
Minor goal was keeping the general structure as this is just slimming down the dependencies of the original code.